jina/peapods/pods/factory.py

```diff
@@ -21,6 +26,13 @@ def build_pod(args: 'Namespace', needs: Optional[Set[str]] = None) -> BasePod:
         :return: the created BasePod
         """
         if getattr(args, 'replicas', 1) > 1:
-            return CompoundPod(args, needs=needs)
+            pod = CompoundPod(args, needs=needs)
         else:
-            return Pod(args, needs=needs)
+            pod = Pod(args, needs=needs)
+
+        if setup_dynamic_routing:
```

Comment:
this should be in the `constructor` of `Pod` and `CompoundPod` themselves

Reply:
This will create conflicts with the ContainerRuntime. If we do this, I would still need to pass the `setup_dynamic_routing`. Are you fine with it being forwarded?

Reply:
Imagine the following test:
```
def test_flow_with_replica_container_ext_yaml(docker_image_built):
    f = Flow().add(
        name='dummyEncoder3',
        uses=f'docker://{img_name}',
        uses_internal=os.path.join(cur_dir, '../../../mwu-encoder/mwu_encoder_ext.yml'),
        parallel=3,
    )

    with f:
        f.index(inputs=random_docs(10))
        f.index(inputs=random_docs(10))
        f.index(inputs=random_docs(10))
```
This should spawn three docker instances insight a CompoundPod. Inside each of the docker instances, you spawn another Pod, which should NOT get the dynamic routing setup. Anyhow, it is spawned via `jina pod` CLI and thus call the `PodFactory.build_pod`. Thus, we need a flip for the `PodFactory.build_pod` method, whether the build pod should use dynamic routing. 

Thinking about it, this again should be handled via the `args` Namespace and potentially only on the actual Pea level. I will create a solution and let you double-check it again.